### ISSUE TITLE
World Browser Modals don't play nicely in GlobalSpace

### ISSUE THREAD

**Issue by TehTurk** [Frooxius/NeosPublic#1348]
So I've made a recreation of the World Browser for People to user in Global Space vs the Users Local Space. This was also to hopefully to be used in the MTC in some Capacity to enable more Social Users who are joining Neos who just want hop in and learn on the fly. 
My information might be a bit lacking as I've had discussions with you about it prior @Frooxius so I'm having some trouble on where the problem I'm having is actually occurring for me to even fix it on my own end. So sorry if I'm unsure of the approach in this situation ^^;

Some issues I've run into are:
- Users Interacting seem to spawn different modals as the information seems to update separately or really inconsistently 
- In some cases it'll spawn completely fine, and others it'll just be the border, or a fuzzed out version of the modal even tho nothing has actually been changed or tweaked to the UI or Material Options, this could be due to Information not populating correctly
- I've tried replacing or even recreating the Modals, but it seems that even using copied over Templates from my actual World Browser seems to not even work correctly.
- I suspect that the WorldBrowser isn't correctly removing my user in some way, or that there's something internal to one of the objects that's affecting the process.
- Even if I have it working remotely, alot of the button operations seem to basically break after the Initial Spawning/Focus of a WorldItem, or the Modal Generates in a weird way requiring me to check on the State of the Modal from each individual item with Logix. (Click Item -> Screen Gets Bigger -> Try to go to any other, multiple things show) or (Click X to leave world, Modal Default shows, but has world information behind it) 

Another thing I've noticed is that you can only have 1 ModalOverlayManager per Canvas unless it's made in a special way, having multiple is somewhat tricky as the Original Solution(World Browser) Seemed to use Internal Values with Constructors which you can't really use, but apparently #1050 was the solution for something like that? I always figured you could tie the Event of a Modal being generated by Dragging the Event into one of the UIX Button Fields for OnPressed etc, to just fire the event.

If you need any information or even my example case feel free, I'm open to suggestions as anytime I'm on Neos its a learning experience :) and you always give really nifty info.

**Comment by Frooxius** [comment]
- What exactly is GlobalSpace? Neos doesn't have such a concept. Do you mean in world with other people?
- Does any of this happen with the official world browser? Generally I don't offer support on user created/hacked content, only on the official ones, a lot of it sounds like things aren't setup properly or things that aren't currently supported in the first place, since you technically can't create modals yourself
- What do you mean by "Users Interacting seem to spawn different modals as the information seems to update separately or really inconsistently"? What exactly is happening with this? Are the modals overlaying each other? Does each user spawn a different type of modal?

**Comment by TehTurk** [comment]
- Sorry I was more so talking World Space with others, Global/World sometimes can interchange lingo wise.
- Yes it does, I decided to go the user content route because the behavior just aren't consistent, or there's issues with it, because it just genuinely doesn't work properly outside of the dash. If I generate a WorldBrowser Facet from Scratch, it operates and generates fine, but interaction is an issue, so trying to add that layer that you've had within the Dash interaction wise.
- The Same Modal is technically spawned, but the visuals and underlying generation operates differently per user.

**Comment by Frooxius** [comment]
- I'd recommend sticking to official lingo rather than inventing/adding new terms, they only increase confusion as I'm not fully sure what you're referring to
- What interactions exactly are an issue when trying to operate the spawned WorldBrowser facet? I don't really offer any support on the user-setup content sorry, but I can have a look at the official one
- How differently they operate per user? What exactly is different?

**Comment by TehTurk** [comment]
- Noted, but at the same time there's no official list/board for said lingo, so it's a bit of bear with each other situation. It's typically more Context Based then specific based so I apologize if it confused you. There's been moments where it's happened to me as well vice versa.
- I understand support on the User Content, but more so on having support so users can use the official content to build out there own devices. 
- Basically try to click on a World Item, and No Type of Modal is Generated. Even tho in the World Browser this is the case.
- Backgrounds, Inner Visual Objects, I've had cases of Only a Border Showing, Modal Appearing but only blurred out but the button Focus/Close Buttons are interactable still. 
- This is only feasible as the Modal from the Dash needs to be copied over and used as a Template in a ModalOverlayManager.

**Comment by Frooxius** [comment]
- There's no official list no, but you should either stick to whatever terminology has been previously used officially or avoid it altogether and explain it more verbosely
- You can build your own devices, but it's also your responsibility to set it up properly. Official development fixes any issues that are with the core behaviors, but I can't control if users setup their content properly or not or trace down issues due to user error
- Did you setup the ModalOverlayManager on  your canvas? It's necessary for this to work correctly
- Can any of this be replicated with official setup?
- This is not officially supported, modals don't work this way and copying them like that is likely to corrupt the system.

**Comment by TehTurk** [comment]
- Noted but still, avoiding it is worse off as it doesn't actually encourage the knowledge of the terms of the situation of content. I welcome you to correct me in the future, because if I make these mistakes is that on me, or on the process? If there was documentation I'd refer to it if I could. I'll try to be more clear in the future.
- Yes the ModalOverlayManager is setup on my Canvas, the Only difference between mine and the WorldBrowser is nothing but the Multiple ModalOverlayManagers on the Canvas for the Dash World Browser
- Yes, I just pulled it and the same behaviors occur. As You need to save the Screens within the Dash for this to work properly. 
- I understand but having a Modal Preset isn't there currently, and requires "User Solutions" , most of this is me learning from the actual object itself.

**Comment by TehTurk** [comment]
Just wanted to say, I felt like I could have worded some of the exchange a bit better. If it helps I will get Video in the Morning to kinda showcase what I'm dealing with if it helps. It should show a bit better of at least what's going on.

**Comment by TehTurk** [comment]
I'm going to close this issue to restart it from scratch to better present the problem as my description confused the actual problem.
